src/routes/teams.rs

```diff
@@ -1,25 +1,275 @@
+) -> Result<HttpResponse, ApiError> {
+    let team_id = info.into_inner().0.into();
+    let current_user = get_user_from_headers(req.headers(), &**pool)
+        .await
+        .map_err(|_| ApiError::AuthenticationError)?;
+
+    let mut transaction = pool
+        .begin()
+        .await
+        .map_err(|e| ApiError::DatabaseError(e.into()))?;
+
+    // Edit Team Member to set Accepted to True
+    TeamMember::edit_team_member(
+        team_id,
+        current_user.id.into(),
+        None,
+        None,
+        Some(true),
+        None,
+        &mut transaction,
+    )
+    .await?;
+
+    transaction
+        .commit()
+        .await
+        .map_err(|e| ApiError::DatabaseError(e.into()))?;
+
+    Ok(HttpResponse::Ok().body(""))
+}
+
+fn default_role() -> String {
+    "Member".to_string()
+}
+
+#[derive(Serialize, Deserialize, Clone)]
+pub struct NewTeamMember {
+    pub user_id: UserId,
+    #[serde(default = "default_role")]
+    pub role: String,
+    #[serde(default = "Permissions::default")]
+    pub permissions: Permissions,
+}
+
+#[post("{id}/members")]
+pub async fn add_team_member(
+    req: HttpRequest,
+    info: web::Path<(TeamId,)>,
+    pool: web::Data<PgPool>,
+    new_member: web::Json<NewTeamMember>,
+) -> Result<HttpResponse, ApiError> {
+    let team_id = info.into_inner().0.into();
+
+    let mut transaction = pool
+        .begin()
+        .await
+        .map_err(|e| ApiError::DatabaseError(e.into()))?;
+
+    let current_user = get_user_from_headers(req.headers(), &**pool)
+        .await
+        .map_err(|_| ApiError::AuthenticationError)?;
+    let team_member =
+        TeamMember::get_from_user_id(team_id, current_user.id.into(), &**pool).await?;
+
+    if let Some(member) = team_member {
+        if member.permissions.contains(Permissions::MANAGE_INVITES)
+            && member.accepted
+            && new_member.role != crate::models::teams::OWNER_ROLE
+        {
+            let new_user = crate::database::models::User::get(member.user_id, &**pool)
+                .await
+                .map_err(|e| ApiError::DatabaseError(e.into()))?
+                .ok_or_else(|| {
+                    ApiError::InvalidInputError("Invalid User ID specified!".to_string())
+                })?;
+            if !member.permissions.contains(new_member.permissions) {
+                return Err(ApiError::InvalidInputError(
+                    "New member permissions are not contained in your permissions.".to_string(),
+                ));
+            }
+
+            let new_id =
+                crate::database::models::ids::generate_team_member_id(&mut transaction).await?;
+            TeamMember {
+                id: new_id,
+                team_id,
+                user_id: new_member.user_id.into(),
+                name: new_user.username,
+                role: new_member.role.clone(),
+                permissions: new_member.permissions,
+                accepted: false,
+            }
```

Comment:
This needs to check if the user is already part of that team and return an error.